lib/LedgerSMB/Budget.pm

```diff
@@ -246,7 +246,7 @@ sub from_input {
              $line->{amount} =   $input->{"credit_$rownum"}
                                - $input->{"debit_$rownum"};
              $line->{credit} = $line->{amount} if $line->{amount} > 0;
-             $line->{debit}  = $line->{amount} * -1 if $line->{amount} < 0;
+             $line->{debit}  = -$line->{amount} if $line->{amount} < 0;
```

Comment:
Sorry.  I viewed the task as removing magic and as just one task.  '-1' is considered magical and so it got tossed.